src/platform/android/BUILD.gn

```diff
@@ -19,21 +19,26 @@ import("${chip_root}/src/platform/device.gni")
 
 import("${build_root}/config/android_abi.gni")
 import("${chip_root}/build/chip/java/rules.gni")
+import("${chip_root}/src/tracing/tracing_args.gni")
 
 assert(chip_device_platform == "android")
 
 source_set("tracing") {
-  sources = [
-    "tracing.cpp",
-    "tracing.h",
-  ]
-
-  deps = [
-    "${chip_root}/src/lib/support",
-    "${chip_root}/src/tracing",
-    "${chip_root}/src/tracing/perfetto:event_storage",
-    "${chip_root}/src/tracing/perfetto:simple_initialization",
-  ]
+  deps = [ "${chip_root}/src/tracing:tracing_buildconfig" ]
```

Comment:
these deps do not seem used if the source files are not compiled?

Why are we not keeping tracing source_set as is, but add a dependency on it based on matter_enable_tracing_support? And the AndroidChipPlatform-JNI is the one that uses build_config.

So we should have: static_library('android') changed:
  - it requires the buildconfig dependency
  - it requires to make the tracing dependency optional. Also is it needed to be excluded? Would link-time optimization not remove the tracing.h/cpp calls if they are not referenced?